pkg/controller/scyllacluster/migrate.go

```diff
@@ -0,0 +1,431 @@
+					if scSpec.ExposeOptions.CQL == nil {
+						return nil
+					}
+					return &scyllav1alpha1.CQLExposeOptions{
+						Ingress: func() *scyllav1alpha1.CQLExposeIngressOptions {
+							if scSpec.ExposeOptions.CQL.Ingress == nil {
+								return nil
+							}
+							return &scyllav1alpha1.CQLExposeIngressOptions{
+								ObjectTemplateMetadata: scyllav1alpha1.ObjectTemplateMetadata{
+									Labels:      scSpec.ExposeOptions.CQL.Ingress.Labels,
+									Annotations: scSpec.ExposeOptions.CQL.Ingress.Annotations,
+								},
+								IngressClassName: scSpec.ExposeOptions.CQL.Ingress.IngressClassName,
+							}
+						}(),
+					}
+				}(),
+				NodeService: func() *scyllav1alpha1.NodeServiceTemplate {
+					if scSpec.ExposeOptions.NodeService == nil {
+						return nil
+					}
+					return &scyllav1alpha1.NodeServiceTemplate{
+						ObjectTemplateMetadata: scyllav1alpha1.ObjectTemplateMetadata{
+							Labels:      scSpec.ExposeOptions.NodeService.Labels,
+							Annotations: scSpec.ExposeOptions.NodeService.Annotations,
+						},
+						Type:                          scyllav1alpha1.NodeServiceType(scSpec.ExposeOptions.NodeService.Type),
+						ExternalTrafficPolicy:         scSpec.ExposeOptions.NodeService.ExternalTrafficPolicy,
+						AllocateLoadBalancerNodePorts: scSpec.ExposeOptions.NodeService.AllocateLoadBalancerNodePorts,
+						LoadBalancerClass:             scSpec.ExposeOptions.NodeService.LoadBalancerClass,
+						InternalTrafficPolicy:         scSpec.ExposeOptions.NodeService.InternalTrafficPolicy,
+					}
+				}(),
+				BroadcastOptions: func() *scyllav1alpha1.NodeBroadcastOptions {
+					if scSpec.ExposeOptions.BroadcastOptions == nil {
+						return nil
+					}
+
+					return &scyllav1alpha1.NodeBroadcastOptions{
+						Nodes: scyllav1alpha1.BroadcastOptions{
+							Type: scyllav1alpha1.BroadcastAddressType(scSpec.ExposeOptions.BroadcastOptions.Nodes.Type),
+							PodIP: func() *scyllav1alpha1.PodIPAddressOptions {
+								if scSpec.ExposeOptions.BroadcastOptions.Nodes.PodIP == nil {
+									return nil
+								}
+								return &scyllav1alpha1.PodIPAddressOptions{
+									Source: scyllav1alpha1.PodIPSourceType(scSpec.ExposeOptions.BroadcastOptions.Nodes.PodIP.Source),
+								}
+							}(),
+						},
+						Clients: scyllav1alpha1.BroadcastOptions{
+							Type: scyllav1alpha1.BroadcastAddressType(scSpec.ExposeOptions.BroadcastOptions.Clients.Type),
+							PodIP: func() *scyllav1alpha1.PodIPAddressOptions {
+								if scSpec.ExposeOptions.BroadcastOptions.Clients.PodIP == nil {
+									return nil
+								}
+								return &scyllav1alpha1.PodIPAddressOptions{
+									Source: scyllav1alpha1.PodIPSourceType(scSpec.ExposeOptions.BroadcastOptions.Clients.PodIP.Source),
+								}
+							}(),
+						},
+					}
+				}(),
+			}
+		}(),
+		// v1.ScyllaCluster doesn't provide API to specify cluster/datacenter wide properties of scylladb.
+		RackTemplate: nil,
+		Racks: func() []scyllav1alpha1.RackSpec {
```

Comment:
nit: maybe some of these big conversion sub-functions could be extracted and uni-tested separately